compiler/testData/compileKotlinAgainstKotlin/noAssertionsForKotlin.kt

```diff
@@ -0,0 +1,28 @@
+// TARGET_BACKEND: JVM
```

Comment:
Looks like this test doesn't check what was tested previously in `testNoAssertionsForKotlinFromBinary`: that there are no assertions on calls to Kotlin declarations from dependencies. I'm afraid we'll have to keep that test until we have the necessary infrastructure to migrate it, namely multi-module bytecode text tests.

Reply:
You're right, there's no way to test here that there are no intrinsic calls. I'll keep this one test in `GenerateNotNullAssertionsTest` and add a comment.

Reply:
I've moved this test to tests-common and at least enabled it for both the JVM and JVM_IR backends, with a TODO to remove the test class once multi-module bytecode text tests become available.